### PR TITLE
fix(exports): retention people name in csvs

### PR DESCRIPTION
**Pull request by thmsobrmlr**
## Problem

People names come up blank in retention modal export. See [here](https://posthog.slack.com/archives/C04UGBUARDG/p1681909516707269) for a customer reported issue.

## Changes

This PR replaces the code for extracting the person identifier with the [name property](https://github.com/PostHog/posthog/blob/c18931e7ff7482d7f6b99518e068a094a6805801/posthog/api/person.py#L90) of the [serialized person](https://github.com/PostHog/posthog/blob/c18931e7ff7482d7f6b99518e068a094a6805801/posthog/queries/actor_base_query.py#L246).

## How did you test this code?

Verified that the first column matches the name displayed in the retention modal.

### CODE REVIEW

File: posthog/tasks/exports/csv_exporter.py
Comment (Twixes) [inline posthog/tasks/exports/csv_exporter.py:118]: This will break for users who do rely on `email` or `id`, right? We just need to use `team.person_display_name_properties` – that's configurable in project settings, so always what the user needs (and aligned between backend and frontend)

**Review by Twixes — DISMISSED** [review]

**Review by benjackwhite — CHANGES_REQUESTED** [review]

File: posthog/tasks/exports/csv_exporter.py
Comment (thmsobrmlr) [reply posthog/tasks/exports/csv_exporter.py:118]: The name property is built with this method:

```python
def get_person_name(person: Person) -> str:
    if person.properties.get("email"):
        return person.properties["email"]
    if len(person.distinct_ids) > 0:
        # Prefer non-UUID distinct IDs (presumably from user identification) over UUIDs
        return sorted(person.distinct_ids, key=is_anonymous_id)[0]
    return person.pk
```
I'm adding the requested changes in a staked PR: https://github.com/PostHog/posthog/pull/15198.

**Comment by thmsobrmlr** [comment]
I just verified that this now takes into account the team's preferred display name

**Review by benjackwhite — APPROVED** [review]

**Comment by benjackwhite** [comment]
Nice! I imagine that was quite the effort to get all the right things in place 🙈

**Comment by thmsobrmlr** [comment]
Yup, but had @Twixes help on the hard parts

**Comment by thmsobrmlr** [comment]
@Twixes Seems I can't merge this without approval from you, once you've already submitted a review :)

**Comment by mariusandra** [comment]
I think you can:
image

**Pull request merged**
